pkg/clients/azure/azure.go

```diff
@@ -0,0 +1,63 @@
+package azure
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/Azure/kubelogin/pkg/token"
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+	"k8s.io/client-go/rest"
+)
+
+// Credentials Secret content is a json whose keys are below.
+const (
+	CredentialsKeyClientID       = "clientId"
+	CredentialsKeyClientSecret   = "clientSecret"
+	CredentialsKeyTenantID       = "tenantId"
+	CredentialsKeyClientCert     = "clientCertificate"
+	CredentialsKeyClientCertPass = "clientCertificatePassword"
+)
+
+func WrapRESTConfig(_ context.Context, rc *rest.Config, credentials []byte, _ ...string) error {
+	m := map[string]string{}
+	if err := json.Unmarshal(credentials, &m); err != nil {
+		return err
+	}
+
+	fs := pflag.NewFlagSet("kubelogin", pflag.ContinueOnError)
+	opts := token.NewOptions()
+	opts.AddFlags(fs)
+	// opts are filled with provided args
```

Comment:
Some more details in the comment would be helpful here for posterity, e.g. we are reading ServerID from flags.